src/portal-impl.c

```diff
@@ -220,14 +476,22 @@ find_portal_implementation (const char *interface)
           if (!g_strv_contains ((const char **)impl->interfaces, interface))
             continue;
 
-          if (g_strv_case_contains ((const char **)impl->use_in, desktops[i]))
+          if (portal_impl_matches_config (impl, interface))
+            {
+              g_debug ("Using %s.portal for %s in %s (config)", impl->source, interface, desktops[i]);
+              return impl;
+            }
+
+          /* Fallback */
+          if (impl->use_in != NULL && g_strv_case_contains ((const char **)impl->use_in, desktops[i]))
             {
-              g_debug ("Using %s for %s in %s", impl->source, interface, desktops[i]);
+              g_debug ("Using %s.portal for %s in %s (fallback)", impl->source, interface, desktops[i]);
```

Comment:
I think this needs to be in a separate loop otherwise you could use a fallback even though we have  a portal that matches the config.

Reply:
Indeed. Done.